packages/payment-detection/test/erc20/fee-proxy-contract.test.ts

```diff
@@ -282,4 +282,61 @@ describe('api/erc20/fee-proxy-contract', () => {
         .feeBalance.balance,
     ).toBe('7');
   });
+
+  it('should have gasFee & gasUsed in the payment eventl', async () => {
+    const mockRequest: RequestLogicTypes.IRequest = {
+      creator: { type: IdentityTypes.TYPE.ETHEREUM_ADDRESS, value: '0x2' },
+      currency: {
+        network: 'rinkeby',
+        type: RequestLogicTypes.CURRENCY.ERC20,
+        value: '0xFab46E002BbF0b4509813474841E0716E6730136', // local ERC20 token
+      },
+      events: [],
+      expectedAmount: '1000000000000000000',
+      extensions: {
+        [ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT]: {
+          events: [
+            {
+              name: 'create',
+              parameters: {
+                feeAddress: '0x35d0e078755Cd84D3E0656cAaB417Dee1d7939c7',
+                feeAmount: '1000000000000000',
+                paymentAddress: '0x6d69c636c825263Aa71a3D86D32D0E4897a7a580',
+                salt: '4f43d8b97ae1f63d',
+              },
+              timestamp: 1655883560,
+            },
+          ],
+          id: ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_FEE_PROXY_CONTRACT,
+          type: ExtensionTypes.TYPE.PAYMENT_NETWORK,
+          values: {
+            // refundAddress: '0xrefundAddress',
+            feeAddress: '0x35d0e078755Cd84D3E0656cAaB417Dee1d7939c7',
+            feeAmount: '1000000000000000',
+            paymentAddress: '0x6d69c636c825263Aa71a3D86D32D0E4897a7a580',
+            salt: '4f43d8b97ae1f63d',
+          },
+          version: '0.2.0',
+        },
+      },
+      extensionsData: [],
+      requestId: '01036165cd1608f6ec52acc6349535c44fec6865cfa930d321394db28a69dfd950',
+      state: RequestLogicTypes.STATE.CREATED,
+      timestamp: 0,
+      version: '0.2',
+    };
+
+    erc20FeeProxyContract = new ERC20FeeProxyPaymentDetector({
+      advancedLogic: mockAdvancedLogic,
+      currencyManager,
+    });
+
+    const balance = await erc20FeeProxyContract.getBalance(mockRequest);
```

Comment:
should I also mock the `extractEvents` result for this test? @benjlevesque

Reply:
if this test is running without mock, I believe you can yes!

don't try to write the mocked data yourself, copy the exact return value from that function 🙏